### PR TITLE
supervisor: Create a file /sd/placeholder.txt with a note

### PR DESCRIPTION
**Pull request by jepler**
sd/placeholder.txt (uses 1 sector):
```
This placeholder file allows mounting an SD card at /sd
```

~~settings.toml (539 bytes, so 2 sectors, if all features enabled):~~ (disabled now due to code size growth)
```toml
# Settings in this file can be retrieved via os.getenv()
# Certain settings starting with CIRCUITPY_ are also used by CircuitPython.

### Core settings
# CIRCUIPY_HEAP_START_SIZE = 8192
# CIRCUITPY_PYSTACK_SIZE = 1536

### Wifi settings
# CIRCUITPY_WIFI_SSID = "wifi network name"
# CIRCUITPY_WIFI_PASSWORD = "wifi password"

### Web workflow settings
# CIRCUITPY_WEB_API_PASSWORD = "api password"
# CIRCUITPY_WEB_API_PORT = 80
# CIRCUITPY_WEB_INSTANCE_NAME = "web instance name"

### BLE settings
# CIRCUITPY_BLE_NAME = "ble device name"
```

These will be created as 0-length files on non-FULL builds.

### CODE REVIEW

**Comment by jepler** [comment]
Even with the new settings.toml text removed, this over-fills a couple of boards, including one where the sd placeholder would be helpful (feather_m0_adalogger). I'll see if it can be pared down further.

**Comment by RetiredWizard** [comment]
I understand that this doesn't really makes sense for boards without sdcardio/sdioio, however those boards do get plugged into carrier boards with SD cards, and sdcard breakouts can also be added on to boards without native sdcard support.

I'm only bringing this up for consideration, I'm really not sure what the answer is but, is the possible confusion from having debatably, two different mounting methods for boards that have native SD card support and those that don't going to be more problematic than the supporting the new method on all boards?

I do really like the idea of having the mount point created and perhaps the use of SD cards on non-native SD Card boards is so tiny that it's not worth worrying about.

**Comment by jepler** [comment]
The "/sd (or other mount point) has to be a directory"  requirement in 9 is for adafruit_sdcard as well as for sdcardio/sdioio. It's just that I'm using this #define to turn the functionality off where it's less likely to fit.

**Comment by bill88t** [comment]
Half a kb for holding placeholder text is not really ideal.
May I instead suggest something along the lines of:
```
# Settings in this file can be retrieved via os.getenv()
# Certain settings starting with CIRCUITPY_ are also used by CircuitPython.
# For more information, visit docs.circuitpython.org/../settings-toml-tutorial
```
Excluding the url that would be only 163 bytes and being presented a nice webpage explaining the options is much more functional in my opinion.

Additionally having `/sd` always exist, a fixed unconfigurable name, while only a small amount of projects will rely on permenant sdcard mounts, doesn't sit right with me.
It only saves a few lines of python code.

This flash space could have easily been redirected to be made into a `isdir`/`isfile`/`pathexists` function.
This would provide a lot more utility than a premade directory at about the same (or less) flash cost.
(these funcs would only require an integer comparison from a `stat()`)

**Comment by tannewt** [comment]
> Additionally having `/sd` always exist, a fixed unconfigurable name, while only a small amount of projects will rely on permenant sdcard mounts, doesn't sit right with me.

It will be created by default but can be deleted. I think this is fine.

File: supervisor/shared/filesystem.c [hunk 1]
```diff
@@ -137,11 +141,19 @@ bool filesystem_init(bool create_allowed, bool force_create) {
         // https://specifications.freedesktop.org/trash-spec/trashspec-latest.html
         #endif
 
+        #if CIRCUITPY_SDCARDIO || CIRCUITPY_SDIOIO
+        res = f_mkdir(&vfs_fat->fatfs, "/sd");
+        #if CIRCUITPY_FULL_BUILD
+        MAKE_FILE_WITH_OPTIONAL_CONTENTS(&vfs_fat->fatfs, "/sd/placeholder.txt",
+            "This placeholder file allows mounting an SD card at /sd\n");
```
Comment (tannewt) [inline supervisor/shared/filesystem.c:148]: The file itself isn't required so I'd say something different.

```suggestion
            "SD cards are mounted at /sd and will shadow this file from Python. USB will not see the SD card contents here.\n");
```

**Review by tannewt — CHANGES_REQUESTED** [review]
One suggestion for the SD placeholder.txt.

File: supervisor/shared/filesystem.c (hunk 1 above)
Comment (jepler) [reply supervisor/shared/filesystem.c]: done!

**Review by tannewt — APPROVED** [review]
Thank you! I wonder if we should translate these strings in the future.

**Comment by jepler** [comment]
I hope it fits this time. :sweat:

File: supervisor/shared/filesystem.c [hunk 2]
```diff
@@ -137,11 +141,20 @@ bool filesystem_init(bool create_allowed, bool force_create) {
...
 
+        #if CIRCUITPY_SDCARDIO || CIRCUITPY_SDIOIO
+        res = f_mkdir(&vfs_fat->fatfs, "/sd");
+        #if CIRCUITPY_FULL_BUILD
+        MAKE_FILE_WITH_OPTIONAL_CONTENTS(&vfs_fat->fatfs, "/sd/placeholder.txt",
+            "SD cards are mounted at /sd and will shadow this file from Python."
+            " USB will not see the SD card contents here.\n");
+        #endif
+        #endif
+
```
Comment (RetiredWizard) [inline supervisor/shared/filesystem.c:152]: I will occasionally mount SD cards to alternate folders so the placeholder message was bothering me a bit. It also took me a couple reads before I understood what the second sentence was trying to say. It's probably just me but what do you think about this as an alternate wording? 

"SD cards mounted at /sd will hide this file from Python."
" SD cards are not visible via USB CIRCUITPY.\n"

Comment (tannewt) [reply supervisor/shared/filesystem.c:152]: I like it!

**Review by tannewt — APPROVED** [review]
Thank you both!

**Pull request merged**